modules/shared/src/main/scala/org/tessellation/currency/dataApplication/package.scala

```diff
@@ -134,6 +136,8 @@ trait DataApplicationL0Service[F[_], D <: DataUpdate, DON <: DataOnChainState, D
     extends DataApplicationService[F, D, DON, DOF]
     with DataApplicationContextualOps[F, D, DON, DOF, L0NodeContext[F]] {
   def genesis: DataState[DON, DOF]
+
+  def onConsumeArtifact(snapshot: Signed[CurrencyIncrementalSnapshot])(implicit G: Applicative[F]): F[Unit] = G.unit
```

Comment:
Please name it "A" as Applicative or just "applicative: Applicative". The same pattern with M: Monad, and so on. F and G may be misleading as F and G are usually used for generic type bounds